env/io_posix.cc

```diff
@@ -56,6 +56,57 @@ int Fadvise(int fd, off_t offset, size_t len, int advice) {
 #endif
 }
 
+// On MacOS (and probably *BSD), the posix write and pwrite calls do not support
+// buffers larger than 2^31-1 bytes. These two wrappers fix this issue by
+// cutting the buffer in 1GB chunks. We use this chunk size to be sure to keep
+// the writes aligned.
+
+bool Write(int fd, const char* buf, size_t nbyte) {
```

Comment:
I'm slightly worry that the function name is too common and it will create conflict in unity build.

Reply:
Named changed to `PosixWrite` (and `PositionedWrite` has been similarly renamed to `PosixPositionedWrite`).